sdk/search/Azure.Search.Documents/src/Models/QueryAnswer.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    /// <summary>
+    /// Configuration for how semantic search returns answers to the search.
+    /// </summary>
+    public partial class QueryAnswer
+    {
+        private const string QueryAnswerCountRaw = "count-";
+        private const string QueryAnswerThresholdRaw = "threshold-";
+
+        /// <summary> A value that specifies whether <see cref="SemanticSearchResults.Answers"/> should be returned as part of the search response. </summary>
+        public QueryAnswerType? AnswerType { get; set; }
```

Comment:
I haven't made `AnswerType` a required field because it is only used in the context of semantic search. I'm creating by default instances [here](https://github.com/Azure/azure-sdk-for-net/blob/6593a9223374af7e946c89d776cc62ad624060e2/sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs#L162) for `SemanticSearchOptions` and [here](https://github.com/Azure/azure-sdk-for-net/blob/6593a9223374af7e946c89d776cc62ad624060e2/sdk/search/Azure.Search.Documents/src/Options/SemanticSearchOptions.cs#L18) for `QueryAnswer`, so that I can take advantage of the generated serialization and deserialization code that accesses [these](https://github.com/Azure/azure-sdk-for-net/blob/6593a9223374af7e946c89d776cc62ad624060e2/sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs#L177-L189) properties.

Additionally, I considered that if a user sets the `AnswerType` property as shown below, they won't encounter a `NullReferenceException`:

```csharp
SearchOptions searchOptions = new SearchOptions();
searchOptions.SemanticSearch.QueryAnswer.AnswerType = QueryAnswerType.Extractive;
```

Reply:
I think we talked in the Arch Board about making the entire thing `null` to indicate you didn't want anything for semantic search was better than having a bunch of nullable properties.  I lean toward the `SemanticSearchOptions SemanticSearch { get; set; } = null;` by default, but don't feel strongly.  It'd be good to run this by @KrzysztofCwalina too.

Reply:
Updated to `SemanticSearchOptions SemanticSearch { get; set; } = null;` by default and made `QueryAnswer.AnswerType`  and `QueryCaption.CaptionType` required.